source/ui/Components/Output/ConsoleSettingsWindow.qml

```diff
@@ -0,0 +1,165 @@
+        if(t==0){
+            ab.checked = true;
+        }
+        else{
+            pl.checked = true;
+        }
+    }
+
+    // The controls for editing console settings.
+    Row {
+        anchors.fill: parent
+        anchors.leftMargin: 15
+        anchors.topMargin: 15
+        anchors.rightMargin: 15
+        anchors.bottomMargin: 15
+
+        spacing: 15
+
+        // Title of each settings control.
+        Column {
+            spacing: 16
+            Text {
+                text: "Memory Source (Address):"
+            }
+            Text {
+                text: "Mode:"
+            }
+        }
+
+        // Settings controls itself.
+        Column {
+            id: controlsColumn
+            spacing: 8
+
+            // Converts a given dec/hex/bin-string to an integer.
+            function integerFromInputString(input) {
+                var base = 10;
+                if (input.indexOf("0x") === 0) {
+                    base = 16;
+                    input = input.slice(2);
+                } else if (input.indexOf("0b") === 0) {
+                    input = input.slice(2);
+                    base = 2;
+                }
+                return parseInt(input, base);
+            }
+
+            // Text field for setting the output item's source address in memory.
+            TextField {
+                id: baseAddressTextField
+
+                onAccepted: { processInput(); }
+                onEditingFinished: { processInput(); }
+
+                // Reads the current input and passes the new value to the model.
+                function processInput() {
+                    var inputValue = controlsColumn.integerFromInputString(String(baseAddressTextField.text))
+                    if (inputValue !== undefined && inputValue >= 0) {
+                        outputComponent.setOutputItemProperty(outputItemIndex, "baseAddress", inputValue);
+                    }
+                }
+            }
+
+            // Text field for settings the number of light-strips.
+            RowLayout {
+                id: mode
+                ExclusiveGroup { id: modeGroup }
+
+                RadioButton{
+                    id:ab
+                    text: "ArrayBased"
+                    checked: true
+                    exclusiveGroup: modeGroup
+                    onClicked: {
+                        mode.processInput();
+                    }
+                }
+                RadioButton {
+                    id: pl
+                    text: "PipeLike"
+                    exclusiveGroup: modeGroup
+                    onClicked: {
+                        mode.processInput();
+                    }
+                }
+
+
+                Component.onCompleted: {
+                    settingsWindowC.updateSettings();
+                }
+
+                // Reads the current input and passes the new value to the model.
+                function processInput() {
+                    var inputValue;
```

Comment:
Could maybe use the ternary operator here: `var inputValue = ab.checked ? 0 : 1;` or even `!ab.checked`